### PR TITLE
add(case crud)

### PR DESCRIPTION
**Pull request by ChrOertlin**
### Added

Adds crud support for the case collection for the monogdb

### Fixed

**Steps to consider while deploying**
- Configuration changes:
- Documentation updates:
- Inform users by email:

### Review:
- [ ] Code approved by
- [ ] Tests executed on stage by:  (Document the test done with screen shots and description.)
- [ ] "Merge and deploy" approved by

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: arnold/crud/create.py [hunk 1]
```diff
@@ -54,5 +55,13 @@ def create_step(adapter: ArnoldAdapter, step: Step) -> List[str]:
     result: InsertOneResult = adapter.step_collection.insert_one(
         step.dict(by_alias=True, exclude_none=True)
     )
-    LOG.info("Updating step %s", step.step_id)
+    LOG.info(f"Updating step {step.step_id}")
+    return result.inserted_id
+
+
+def create_case(case: Case, adapter: ArnoldAdapter):
```
Comment (islean) [inline arnold/crud/create.py:62]: Add type hint

File: arnold/crud/read/case.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+"""CRUD module to find entries."""
+
+
+from arnold.adapter import ArnoldAdapter
+from arnold.models.database.case import Case
+
+
+def get_case(case_id: str, adapter: ArnoldAdapter) -> Case | None:
+    """Retrieve a case document from the database."""
```
Comment (islean) [inline arnold/crud/read/case.py:9]: Maybe mention that it raises a Pydantic error if the case can not be parsed correctly.

File: tests/conftest.py [hunk 3]
```diff
@@ -10,16 +10,23 @@
...
+    database = mongo_client[DATABASE]
+    return ArnoldAdapter(database.client, db_name=DATABASE)
 
+
+@pytest.fixture()
+def mock_adapter() -> ArnoldAdapter:
+    """Return a mock adapter for testing."""
     mongo_client = MongoClient()
     database = mongo_client[DATABASE]
     return ArnoldAdapter(database.client, db_name=DATABASE)
```
Comment (islean) [inline tests/conftest.py:25]: So these are basically the same but one is a fixture while the other is not? Can mock_adapter just be `return override_arnold_adapter()`?

File: tests/crud/test_case.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+def test_create_case(mock_adapter: ArnoldAdapter, balsamic_case_json: dict):
+    """Test creating a case in the database."""
+    # GIVEN a case document
+    case: Case = Case.model_validate(balsamic_case_json)
+
+    # WHEN creating a case in the database
+    create_case(case=case, adapter=mock_adapter)
+
+    # THEN a case is added
+    new_case: Case = mock_adapter.case_collection.find_one()
```
Comment (islean) [inline tests/crud/test_case.py:17]: Would love some more specific way of fetching this than getting the first available case.

**Review by islean — APPROVED** [review]
Would love some additional tests, e.g.  something which breaks if you do not start out with an empty db and something that tests what happens when something goes wrong.d

File: tests/conftest.py (hunk 3 above)
Comment (ChrOertlin) [reply tests/conftest.py:25]: The override arnold adapter is used to override a dependency in the fastapi testclient. So while the code is identical their function is not. I tested with letting it return one or the other but it breaks the tests. Will leave it as is for now.

**Pull request merged**